### PR TITLE
Add check index operation

### PR DESCRIPTION
**Pull request by Smosker**
Checking index status after creation - if its invalid - deleting it and raising error

### CODE REVIEW

File: zero_downtime_migrations/backend/schema.py [hunk 1]
```diff
@@ -240,9 +243,10 @@ def get_query_result(self, sql, params=(), row_count=False):
             return cursor.fetchone()
 
     def parse_cursor_result(self, cursor, place=0, collect_sql_value=1,):
+        result = None
         if self.collect_sql:
             result = collect_sql_value  # For sqlmigrate purpose
-        else:
+        elif cursor:
```
Comment (antonyc) [inline zero_downtime_migrations/backend/schema.py]: this code is not clear for me. Why is cursor may not cast to "true" value?

File: zero_downtime_migrations/backend/schema.py [hunk 2]
```diff
@@ -334,21 +340,57 @@ def _create_unique_constraint_from_index_sql(self, table, index_name):
...
         exit_atomic = False
         # Account for non-string statement objects.
         sql = str(sql)
 
-        if re.search('CREATE.+INDEX', sql):
+        if re.search('(CREATE|DROP).+INDEX', sql):
             exit_atomic = True
             if 'CONCURRENTLY' not in sql:
-                sql = sql.replace('CREATE INDEX', 'CREATE INDEX CONCURRENTLY')
+                sql = sql.replace('INDEX', 'INDEX CONCURRENTLY')
```
Comment (antonyc) [inline zero_downtime_migrations/backend/schema.py]: CONCURRENTLY will always be forced? this should be mentioned in a few words in README.

File: zero_downtime_migrations/backend/schema.py [hunk 3]
```diff
@@ -334,21 +340,57 @@ def _create_unique_constraint_from_index_sql(self, table, index_name):
...
+            if not self._create_unique_failed(exc):
+                raise
+
+        if exit_atomic and not self.collect_sql:
+            invalid_index_name = self._check_valid_index(sql)
+            if invalid_index_name:
+                # index was build, but invalid, we need to delete it
+                self.execute(self.sql_delete_index % {'name': invalid_index_name})
+                raise InvalidIndexError(
+                    'Unsuccessful attempt to create an index, fix data if needed and restart.'
```
Comment (antonyc) [inline zero_downtime_migrations/backend/schema.py]: How data can be fixed by user? Is that obvious? For me it's not.

File: zero_downtime_migrations/backend/schema.py (hunk 2 above)
Comment (Smosker) [reply zero_downtime_migrations/backend/schema.py]: add info to readme

File: zero_downtime_migrations/backend/schema.py (hunk 3 above)
Comment (Smosker) [reply zero_downtime_migrations/backend/schema.py]: it is mentioned in postgresql wiki, add some info to readme

File: zero_downtime_migrations/backend/schema.py (hunk 1 above)
Comment (Smosker) [reply zero_downtime_migrations/backend/schema.py]: it is not the cursor itself but a result of cursor execution, bad naming, renamed

**Pull request merged**